gobblin-runtime/src/main/java/gobblin/runtime/MysqlDatasetStateStore.java

```diff
@@ -51,9 +70,20 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(MysqlDatasetStateStore.class);
 
+  private static final String SELECT_JOB_STATE_LATEST_BY_PREFIX_TEMPLATE =
+          "SELECT state, CASE WHEN locate(?, table_name) = 0 THEN '' "
```

Comment:
Well, if that is the case, this is really tricky.  I think we need to figure out how to deal with this as I believe that using the "current" alias causes a lot of trouble, especially since creating the state and the alias are not an atomic operation.  This can lead to scenarios where the state has been updated and the alias has not been created. Adding a restriction on the dataset URN might be hard because it is already being used and we don't know whether there are values that contain `-`.

@htran1  Do you have any suggestions or ideas?